html/mycakeapp/src/Controller/RatingsController.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace App\Controller;
+
+use App\Controller\AppController;
+
+/**
+ * Ratings Controller
+ *
+ * @property \App\Model\Table\RatingsTable $Ratings
+ *
+ * @method \App\Model\Entity\Rating[]|\Cake\Datasource\ResultSetInterface paginate($object = null, array $settings = [])
+ */
+class RatingsController extends AppController
+{
+    /**
+     * Index method
+     *
+     * @return \Cake\Http\Response|null
+     */
+    public function index()
+    {
+        $this->paginate = [
+            'contain' => ['Biditems'],
+        ];
+        $ratings = $this->paginate($this->Ratings);
+
+        $this->set(compact('ratings'));
+    }
+
+    /**
+     * View method
+     *
+     * @param string|null $id Rating id.
+     * @return \Cake\Http\Response|null
+     * @throws \Cake\Datasource\Exception\RecordNotFoundException When record not found.
+     */
+    public function view($id = null)
+    {
+        $rating = $this->Ratings->get($id, [
+            'contain' => ['Biditems'],
+        ]);
+
+        $this->set('rating', $rating);
+    }
+
+    /**
+     * Add method
+     *
+     * @return \Cake\Http\Response|null Redirects on successful add, renders view otherwise.
+     */
+    public function add()
+    {
+        // すでに同じ、出品者、落札者が入っていないか調査
+        $ratings = $this->Ratings->find('all')->toArray();
+        $is_rating = false;
+        for ($i = 0; $i < count($ratings); $i++) {
+            if ($ratings[$i]["biditem_id"] === intval($_GET["biditem_id"]) && $ratings[$i]["target"] === intval($_GET["target"]) && $ratings[$i]["rater"] === intval($_GET["rater"])) {
+                $is_rating = true;
+                break;
+            }
+        }
```

Comment:
$is_rating (これはレーティングである) という変数名では何を表すのか分からないので変数名NGです！
せめて何を表すのかソースコードにコメントを書いてください。

このガードは既にレビューが登録済みならリダイレクトするという事でしょうが、フラグを導入すると複雑になるのでこのように早期リターンにしましょう。

```suggestion
        for ($i = 0; $i < count($ratings); $i++) {
            if ($ratings[$i]["biditem_id"] === intval($_GET["biditem_id"]) && $ratings[$i]["target"] === intval($_GET["target"]) && $ratings[$i]["rater"] === intval($_GET["rater"])) {

                return $this->redirect(['action' => '../auction']);
            }
        }
```